packages/cli/src/webhooks/webhook.service.ts

```diff
@@ -118,4 +133,212 @@ export class WebhookService {
 			.find({ select: ['method'], where: { webhookPath: path } })
 			.then((rows) => rows.map((r) => r.method));
 	}
+
+	async createWebhookIfNotExists(
+		workflow: Workflow,
+		webhookData: IWebhookData,
+		mode: WorkflowExecuteMode,
+		activation: WorkflowActivateMode,
+	): Promise<void> {
+		const webhookExists = await this.runWebhookMethod(
+			'checkExists',
+			workflow,
+			webhookData,
+			mode,
+			activation,
+		);
+		if (!webhookExists) {
+			// If webhook does not exist yet create it
+			await this.runWebhookMethod('create', workflow, webhookData, mode, activation);
+		}
+	}
+
+	async deleteWebhook(
+		workflow: Workflow,
+		webhookData: IWebhookData,
+		mode: WorkflowExecuteMode,
+		activation: WorkflowActivateMode,
+	) {
+		await this.runWebhookMethod('delete', workflow, webhookData, mode, activation);
+	}
+
+	private async runWebhookMethod(
+		method: WebhookSetupMethodNames,
+		workflow: Workflow,
+		webhookData: IWebhookData,
+		mode: WorkflowExecuteMode,
+		activation: WorkflowActivateMode,
+	): Promise<boolean | undefined> {
+		const node = workflow.getNode(webhookData.node);
+
+		if (!node) return;
+
+		const nodeType = this.nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
+
+		const webhookFn = nodeType.webhookMethods?.[webhookData.webhookDescription.name]?.[method];
+		if (webhookFn === undefined) return;
+
+		const context = new HookContext(
+			workflow,
+			node,
+			webhookData.workflowExecuteAdditionalData,
+			mode,
+			activation,
+			webhookData,
+		);
+
+		return (await webhookFn.call(context)) as boolean;
+	}
+
+	/**
+	 * Executes the webhook data to see what it should return and if the
+	 * workflow should be started or not
+	 */
+	async runWebhook(
+		workflow: Workflow,
+		webhookData: IWebhookData,
+		node: INode,
+		additionalData: IWorkflowExecuteAdditionalData,
+		mode: WorkflowExecuteMode,
+		runExecutionData: IRunExecutionData | null,
+	): Promise<IWebhookResponseData> {
+		const nodeType = this.nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
+		if (nodeType.webhook === undefined) {
+			throw new ApplicationError('Node does not have any webhooks defined', {
+				extra: { nodeName: node.name },
+			});
+		}
+
+		const context = new WebhookContext(
+			workflow,
+			node,
+			additionalData,
+			mode,
+			webhookData,
+			[],
+			runExecutionData ?? null,
+		);
+
+		return nodeType instanceof Node
+			? await nodeType.webhook(context)
+			: ((await nodeType.webhook.call(context)) as IWebhookResponseData);
```

Comment:
same here. I want to fix this by making sure that these functions are infered as `CallableFunction` instead of `Function`, but so far I could not figure this out.